components/list/list.js

```diff
@@ -175,6 +188,35 @@ class List extends SelectionMixin(LitElement) {
 		return new SelectionInfo(keys, selectionInfo.state);
 	}
 
+	_getItemByIndex(index) {
+		const items = this.getItems();
+		if (index > items.length - 1) return;
+		return items[index];
+	}
+
+	async _getItemsShowingCount() {
+		if (this.slot === 'nested') return this._itemsShowingCount;
+		else return this._getListItemsShowingTotalCount(false);
+	}
+
+	_getLastItemIndex() {
+		return this._itemsShowingCount - 1;
+	}
+
+	async _getListItemsShowingTotalCount(refresh) {
```

Comment:
This helper method attempts to reduce the cost of calculating the root showing count by using a cached valued when possible, thereby avoiding some tree walking. 

At first I was going to have the consumer provide this value directly to the `d2l-pager-load-more` component, but future expand/collapse functionality will affect this count so that it will only include items in expanded items (no items in collapsed lists), and would require the consumer to maintain that count based on user interaction with the list. The list is in the best position to do that, but unfortunately needs to deal with the complexity of getting the total root count including nested lists.